### PR TITLE
Robotic console warmup and cooldown

### PR DESCRIPTION
**Pull request by Kubisopplay**
## About The Pull Request
Makes robotic console less op in dealing with cyborgs, without removing its ability to cripple malfunctions, and destroy emagged cyborgs. Hackmd with my thought process: https://hackmd.io/EE9s9BWdSXOi2wpOy8WKtg 
Detonation now requires 60 seconds of lockdown before its possible, with the exception of silicons detonating silicons. Unlocking now takes 30 seconds if done via robotics console, to make it still a 30 second stun, and to require quick reactions to stop detonation. It also gives 60 second cooldown between remote lockdowns, to prevent spamming, and to prevent it from permament lockdown.
Draft for now, since i still need to check if my monkey brain didnt leave a stupid mistake in there, since i checked almost everything
## Why It's Good For The Game
Being able to remotely thanos snap all cyborgs on z-level wasnt very fun for the cyborg players, ais, and anyone in victimity of the cyborg. Leaves the console as strong weapon against rogue ais, but not the absolute death sentence it was before for malfs

## Changelog
:cl:
balance: makes robotic console have a warm up period for cyborg detonation, and makes remote unlocking take 30 seconds
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
As much as I revile the thanos snapping of borgs, I have a bad feeling this makes the already fairly powerful borgs, especially in the hands of a Malf AI, all that stronger even if the lockdown is absolutely the better option for most borg-related issues.

The thing I'm thinking of here is mostly that AIs can also access that console, and since beating a Malf can hinge on getting to that console first to start slamming your head on the big red button to take out all their hands at once.

This is why borg balance hinging on the console is so frustrating for everyone involved, and extremely binary gameplay. I don't think, long term, it is a healthy mechanic for borgs or something worth doubling down on. Rather, I think handling borgs shouldn't require the console at all.

**Comment by ATH1909** [comment]
but detonating an emagged cyborg that a traitor is riding to instantly kill the borg's rider is funny

**Comment by zxaber** [comment]
Cutting cameras in the RD's office will make the AI helpless to stop you, outside preplanning using mid-game tech (engiborg circuit manipulator upgrade, which requires humanoids to install) or just stealing the console (which is obvious as hell).

Seeing as there's no extra delay in locking over detonations, the ability to lock all the borgs and then disable the console with a screwdriver before the AI can unlock them is potentially a decent crew buff against Malf AIs. Unsynced borgs are, after all, the *greatest* threat to an AI (outside ion rifle spam, I suppose). That opportunity is diminished when you detonate all the existing borgs for fear the AI will unlock them before you can do anything.

**Comment by Kubisopplay** [comment]
> Cutting cameras in the RD's office will make the AI helpless to stop you, outside preplanning using mid-game tech (engiborg circuit manipulator upgrade, which requires humanoids to install) or just stealing the console (which is obvious as hell).
> 
> Seeing as there's no extra delay in locking over detonations, the ability to lock all the borgs and then disable the console with a screwdriver before the AI can unlock them is potentially a decent crew buff against Malf AIs. Unsynced borgs are, after all, the _greatest_ threat to an AI (outside ion rifle spam, I suppose). That opportunity is diminished when you detonate all the existing borgs for fear the AI will unlock them before you can do anything.

It is a minor buff, since good robos and rds already did it with current console. Meanwhile it actually lets the better prepared malf, the ones with upgraded engis and stuff not die horribly because someone printed a console out of suspicion, or because CE or RD remebered they have a techfab in locker. 
Most importantly, it cuts on the sudden explosion thing, so borgs can at least scream, say their goodbyes, or in case of actual mistake, which happens more often then you would think.

**Comment by Kubisopplay** [comment]
> As much as I revile the thanos snapping of borgs, I have a bad feeling this makes the already fairly powerful borgs, especially in the hands of a Malf AI, all that stronger even if the lockdown is absolutely the better option for most borg-related issues.
> 
> The thing I'm thinking of here is mostly that AIs can also access that console, and since beating a Malf can hinge on getting to that console first to start slamming your head on the big red button to take out all their hands at once.
> 
> This is why borg balance hinging on the console is so frustrating for everyone involved, and extremely binary gameplay. I don't think, long term, it is a healthy mechanic for borgs or something worth doubling down on. Rather, I think handling borgs shouldn't require the console at all.

You have enough tools outside of console that are as strong, so its extremely minor buff when veterans fight against veterans. Nothing can be done to help newbies fighting against veterans, since you cant whisper tactics to both sides at once. And well, beating malf doesnt need the console, i would say that only without the console playing field is truly level

**Comment by InsaneRed** [comment]
cutting the cams in rd office would be metagaming and by the time ai decides to go loud the cosnole will alraedy have been deconsructed this only buffs dumb ais nothing else

**Comment by InsaneRed** [comment]
Not only this but from what ive seen borgs tend to make door-fortresses around the sat where inner of the sat is just 100 doors, this + unkillable borgs is a cancer no man should have to deal with. borgs are already given alot of buffs with some debuffs to balance them out biggest being you are already immune to the plasma burning around you

**Comment by InsaneRed** [comment]
also this completetly destroys any chance antags have at blowing borgs for whatever reason they may have

**Comment by InsaneRed** [comment]
also just cutting cameras wont work as ais can reactivate them, simply deconstructing  the console will usually delay the issue while you are flooding the other rooms that you can get them, in the end this is just one big skill issue

**Comment by InsaneRed** [comment]
on another combined note, imagine an ai hiding himself somewhere thats hard to find like in space, and just freely flooding, if you make this into a feature it will literlaly make ai too op because he can just fuck off hide in space, simply block any explosions of borgs, and just keep on flooding and then go delta after most of the station is dead (most players dont know the nuke disk points to ai) and simply get a free win. this is a completely unnecesarry buff that ai wouldnt need if he knew how to play anyway

**Comment by Kubisopplay** [comment]
> Not only this but from what ive seen borgs tend to make door-fortresses around the sat where inner of the sat is just 100 doors, this + unkillable borgs is a cancer no man should have to deal with. borgs are already given alot of buffs with some debuffs to balance them out biggest being you are already immune to the plasma burning around you

Oh no, i need to use other counters to borgs then completely unfunny Thanos snap, oh god i need to flash them, how will I live. Sorry insanered, but it's skill issue on your side

**Comment by InsaneRed** [comment]
lol i barely use rd console to kill borgs but sure, anyawy you havent deflected any of my arguments. borgs are given so many positives and  they are meant to be easy to kill taking away and making the rd console useless is not the way to go about this if you wanna change something change their insta stun and in return you can take away laser pointers and other options (cyborgs should just be removed imo but whatever)

**Comment by Sparkezel** [comment]
Immune to radiation
Immune to pressure
Immune to space
Immune to heat
Immune to fire
No damage slowdown
Many tools
Old baton
>boo hoo roboconsole too op :( flash too

**Comment by Kubisopplay** [comment]
> Immune to radiation
> Immune to pressure
> Immune to space
> Immune to heat
> Immune to fire
> No damage slowdown
> Many tools
> Old baton
> 
> > boo hoo roboconsole too op :( flash too

Going with the same style
AOE/ranged hits can/point and click/ranged AOE hardstuns
Radiation melts them with damage they can barely repair
Instant remote thanos snap
Automatic suspicion of everyone on station even when fully loyal and asimov

Out of this all i am just changing instant Thanos snap to instant remote guaranteed 30 second stun every minute

**Comment by Kubisopplay** [comment]
> lol i barely use rd console to kill borgs but sure, anyawy you havent deflected any of my arguments. borgs are given so many positives and they are meant to be easy to kill taking away and making the rd console useless is not the way to go about this if you wanna change something change their insta stun and in return you can take away laser pointers and other options (cyborgs should just be removed imo but whatever)

I have some ideas for that, and it would be a revamp, but that's something for after feature freeze. Probably will get screamed at because "borgs op", completely ignoring fact that humans can easily prepare for anything borgs are immune, and even more

**Comment by Sparkezel** [comment]
borgs also have large interaction range so they could lock and bolt + shock your airlocks

**Comment by Kubisopplay** [comment]
> borgs also have large interaction range so they could lock and bolt + shock your airlocks

So do lasers, EMP guns, laser pointers, and even the nerfed robo console

**Comment by InsaneRed** [comment]
buffing something already powerful is not a good idea, this is unnecessary try to remove laser pointers not this

**Comment by InsaneRed** [comment]
and again you still havent responded to how this considers traitors or literally any other antag

File: tgui/packages/tgui/interfaces/RoboticsControlConsole.js [hunk 1]
```diff
@@ -70,16 +70,22 @@ const Cyborgs = (props, context) => {
...
-              color="bad"
-              onClick={() => act('killbot', {
+              disabled={!cyborg.locked_down&&!cyborg.can_blow}
+              content={!cyborg.can_blow&&cyborg.locked_down
+                ?"Arming up"
+                :"Detonate"}
+              color={cyborg.can_blow ? "bad"
+                :cyborg.locked_down ? "yellow" : "grey"}
+              onClick={() => act(cyborg.can_blow
+                ?'killbot' : "", {
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/RoboticsControlConsole.js:88]: Must be properly formatted, use `bin/tgui --fix`

Comment (Kubisopplay) [reply tgui/packages/tgui/interfaces/RoboticsControlConsole.js:88]: I used the command, it changed nothing. Could you elaborate on what i need to fix in that case?

**Comment by Burgermancoding** [comment]
The counter is to decon these consoles and depower rooms. This is pointless and a far overreaching nerf. Silicons can wordlessly pull off mass murder in many ways without this being a problem.

**Comment by Fikou** [comment]
cant i just screwdriver the console to break it and screwdriver it again for fresh console with no cooldown

**Comment by JoshAdamPowell** [comment]
oh look guys Microsoft is making an AI buff!

**Comment by Kubisopplay** [comment]
> cant i just screwdriver the console to break it and screwdriver it again for fresh console with no cooldown

No, the cooldown is counted on the borg.

> The counter is to decon these consoles and depower rooms. This is pointless and a far overreaching nerf. Silicons can wordlessly pull off mass murder in many ways without this being a problem.

The "counter" that you said is not a counter, since it doesnt matter if you decon the console, when you can literally create  new one in 30 seconds. This nerf is not pointless, since it mostly empowers normal asimov ais, and malfs that dont want to go with the meta way of flooding the shit out of station, since any other strategy requires relying on cyborgs, where not only you can never rely on forceborged people, but one competent person with sci access means you lost without counterplay. Please ask anyone who played ai for longer how does it look.

**Comment by Burgermancoding** [comment]
I think you missed the point where I said depower the rooms, if you just depower the rooms that can print them or have them you win. Or you blow them up. Or you get a borg to do that before you go loud. This is just a thinly veiled attempt to make the game easier for YOURSELF personally

**Comment by Kubisopplay** [comment]
> I think you missed the point where I said depower the rooms, if you just depower the rooms that can print them or have them you win. Or you blow them up. Or you get a borg to do that before you go loud. This is just a thinly veiled attempt to make the game easier for YOURSELF personally

Considering i stopped maining AI a long time ago, that my idea was supported by every ai main i talked to with exception of insanered, and that everyone just says "uh oh you can decon" without remembering how does it look... No, i am not trying to make it easier for myself, i am trying to make malf not only about delta, and not about dying cause of one person who decided that ai looks suspicious and to print console board for himself.

Also funny how you all ignore the fact that this is mostly QoL for cyborgs themselves, since locked borgs can be unsynced/fixed, while explosion either wastes 10 minutes of your life and a lot of resources, or just flat out removes you from the game, without any warning nor counterplay.


> and again you still havent responded to how this considers traitors or literally any other antag

I am still thinking about making it possible to emag the console to be able to detonate instantly, or on much shorter cooldown.

**Comment by Burgermancoding** [comment]
Supported by players who want the game to be easier for them regardless of the fact they already have many abilities that people playing other roles don't have is not a good argument, if you are getting your borgs blown you have misplayed somehow and its your own fault, If it is not a misplay then it is for very poor reasons then you ahelp and get the people blowing your borgs up fnr banned, or if it's a traitor then that's just a shit out of luck situation for the player. You forget that silicons are the most annoying thing to be on the receiving end of when you're a player because you can't fight someone who is fucking with you by and large remotely, or is instantly stunning you, locking you in and shocking you etc, this will just add to player frustrations and lead to borgs/silicons receiving bigger nerfs down the line, so i'd take that into consideration before you merge this

**Comment by Tekkuni** [comment]
The lockdown period before being able to detonate a borg is **definitely** something that should be in so people don't detonate borgs as a kneejerk reaction, which often happens when borgs are 100% legit, and are instead encouraged to lock the borg down and actually inspect it.

The delta t between time spent getting unlocked and being able to be locked down again is the biggest question here. All it takes is one organic dedicated to a borg console, and the AI having a console of their own, and all borgs will be in the cycle of being locked and unlocked for all eternity. I'd probably rather see giving them a grace period as something that costs processing time for malf, and just isn't doable for subverted AIs so it's on the AI alone to find and disable the robo console, assuming their borgs are not emagged. Would really need to see the grace period in play to be fair - is it enough to kill an organic if you find one quick, should it even be enough to kill one? Why have one if it isn't? Being found locked is a death sentence, 100%, so the vulnerability is still very real.

**Comment by Kubisopplay** [comment]
Burgerman, removing something that is impossible to counterplay, and is just an outcome known as "lag and pink text telling you that you died" is not a big nerf. The grace period is something i wasnt sure how to implement, and would gladly take the suggestions. Also, i have few ideas that i wanted to explore for making silicons less cancerous to fight, but considering the reaction here it makes no sense to even try. About the missplay thing... tell me, how do you want to instantly and stealthy secure 7 points on the station, 3 out of them are out in the open, and missing any one of them can be instant death to all cyborgs on z-level?

**Comment by Kubisopplay** [comment]
From the ideas on discord, making the console instant, but only able to remove the remote abilities of cyborgs, like shocking bolting and opening doors, and maybe turning off the modules, or remotely slowing the cyborg down

**Comment by InsaneRed** [comment]
this doesnt in any help make making malf not only delta this literally buffs borgs for no reason at all,borgs have th eadvantage of remote acess to most statino assets, being able to flood half the station in an instant , being immune to wounds,oxy,(almost all)burn damage, and the ability to heal themseelves + a variety of tools this is simply not the way to go if you wanna "make ai not only delta" ai is flawed in itself game mechanic wise, not letting us kill borgs because ai failed to do what he is supposed to do is not a solution to that. try adding more ai malf abilities

**Comment by InsaneRed** [comment]
the whole idea of the robo console is its deadly for malf ais that dont secure it, this will only make it mor eannoying to deal with the threat that is already burning down everything near you not to say emagging and forcing a traitor to waste tc that he only has limited on is retarded and shouldnt be how it is (especially with the new biddle traitor stuff that will probably get merged)

**Pull request closed without merging**